packages/integrations/image/components/index.ts

```diff
@@ -1,16 +1,57 @@
 /// <reference types="astro/astro-jsx" />
 export { default as Image } from './Image.astro';
 export { default as Picture } from './Picture.astro';
+import type { HTMLAttributes as AllHTMLAttributes } from '../../../astro/types';
 
-// TODO: should these directives be removed from astroHTML.JSX?
-export type ImgHTMLAttributes = Omit<
-	astroHTML.JSX.ImgHTMLAttributes,
-	'client:list' | 'set:text' | 'set:html' | 'is:raw'
->;
-export type HTMLAttributes = Omit<
-	astroHTML.JSX.HTMLAttributes,
-	'client:list' | 'set:text' | 'set:html' | 'is:raw'
->;
+import type { TransformOptions, OutputFormat } from '../dist/loaders/index.js';
+import type { ImageMetadata } from '../dist/vite-plugin-astro-image.js';
+
+export interface ImageComponentLocalImageProps
+	extends Omit<TransformOptions, 'src'>,
+		Omit<ImgHTMLAttributes, 'src' | 'width' | 'height'> {
+	src: ImageMetadata | Promise<{ default: ImageMetadata }>;
+	/** Defines an alternative text description of the image. Set to an empty string (alt="") if the image is not a key part of the content (it's decoration or a tracking pixel). */
+	alt: string;
+}
+
+export interface ImageComponentRemoteImageProps extends TransformOptions, ImgHTMLAttributes {
+	src: string;
+	/** Defines an alternative text description of the image. Set to an empty string (alt="") if the image is not a key part of the content (it's decoration or a tracking pixel). */
+	alt: string;
+	format?: OutputFormat;
+	width: number;
+	height: number;
+}
+
+export interface PictureComponentLocalImageProps
+	extends Omit<HTMLAttributes, 'src' | 'width' | 'height'>,
+		Omit<TransformOptions, 'src'>,
+		Pick<ImgHTMLAttributes, 'loading' | 'decoding'> {
+	src: ImageMetadata | Promise<{ default: ImageMetadata }>;
+	/** Defines an alternative text description of the image. Set to an empty string (alt="") if the image is not a key part of the content (it's decoration or a tracking pixel). */
+	alt: string;
+	sizes: HTMLImageElement['sizes'];
+	widths: number[];
+	formats?: OutputFormat[];
+}
+
+export interface PictureComponentRemoteImageProps
+	extends Omit<HTMLAttributes, 'src' | 'width' | 'height'>,
+		TransformOptions,
+		Pick<ImgHTMLAttributes, 'loading' | 'decoding'> {
+	src: string;
+	/** Defines an alternative text description of the image. Set to an empty string (alt="") if the image is not a key part of the content (it's decoration or a tracking pixel). */
+	alt: string;
+	sizes: HTMLImageElement['sizes'];
+	widths: number[];
+	aspectRatio: TransformOptions['aspectRatio'];
+	formats?: OutputFormat[];
+	background: TransformOptions['background'];
+}
+
+export type ImgHTMLAttributes = AllHTMLAttributes<'img'>;
+
+export type HTMLAttributes = Omit<astroHTML.JSX.HTMLAttributes, 'set:text' | 'set:html' | 'is:raw'>;
```

Comment:
In my opinion, it'd be better to leave the `astro/types` `HTMLAttributes` as is, without renaming it. And this one can be called `GlobalHTMLAttributes`, since it's only the global ones that are included here

Additionally, the list of exclusions can be built from Astro's types, like done in `astro/types`:

```ts
export type HTMLAttributes<Tag extends HTMLTag> = Omit<
	astroHTML.JSX.IntrinsicElements[Tag],
	keyof Omit<AstroBuiltinAttributes, 'class:list'>
>;
```